source/solution_deploy/source/action_target_provider.py

```diff
@@ -0,0 +1,152 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Custom resource provider for AWS Security Hub Custom Action Target"""
+
+# Test Event
+# {
+#     "ResourceProperties": {
+#         "Name": "Remediate with ASR",
+#         "Description": "Submit the finding to Automated Security Response on AWS",
+#         "Id": "ASRRemediation"
+#     },
+#     "RequestType": "Create",
+#     "ResponseURL": "https://bogus"
+# }
+
+import os
+import json
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+from logger import Logger
+import cfnresponse
+
+# initialize logger
+LOG_LEVEL = os.getenv("log_level", "info")
+logger_obj = Logger(loglevel=LOG_LEVEL)
+REGION = os.getenv("AWS_REGION", "us-east-1")
+PARTITION = os.getenv("AWS_PARTITION", default="aws")  # Set by deployment template
+
+BOTO_CONFIG = Config(retries={"mode": "standard"})
+CLIENTS = {}
+
+
+def get_securityhub_client():
+    if "securityhub" not in CLIENTS:
+        CLIENTS["securityhub"] = boto3.client("securityhub", config=BOTO_CONFIG)
+    return CLIENTS["securityhub"]
+
+
+class InvalidCustomAction(Exception):
+    pass
+
+
+class CustomAction(object):
+    """
+    Security Hub CustomAction class
+    """
+
+    name = ""
+    description = ""
+    id = ""
+    account = ""
+
+    def __init__(self, account, properties):
+        self.name = properties.get("Name", "")
+        self.description = properties.get("Description", "")
+        self.id = properties.get("Id", "")
+        self.account = account
+        if not self.name or not self.description or not self.id:
```

Comment:
if we use TypeHints for the input params, we can help the caller not to pass in empty strings or null values in the first place